framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -317,3 +317,19 @@ MaterialPropertyInterface::checkExecutionStage()
     mooseError("Material properties must be retrieved during object construction. This is a code "
                "problem.");
 }
+
+const std::vector<MaterialPropertyInterface *>
+MaterialPropertyInterface::getInterfaceObjects(MooseApp & app)
```

Comment:
This was part of my initial design, but once you have one object inheriting from several interfaces that do something like this it gets ugly real quick

Reply:
```
MaterialPropertyInterface::getMaterialPropertyInterfaceObjects
```

:-)

Reply:
But, yeah, I'm sure that can be refined once we get to that point